engine/src/main/java/org/destinationsol/game/screens/InventoryScreen.java

```diff
@@ -77,7 +108,23 @@ public void blurCustom(SolApplication solApplication) {
         }
     }
 
+    public void refresh() {
+        itemList.recalculate();
```

Comment:
You shouldn't have to recalculate this every single frame. Something feels off.

Note that recalculate on itemList will potentially trigger recalculate on every single UI element on the screen, so using it sparingly is wise.

Reply:
It is not recalculated every frame, although it used to be. The UiItemList class was been modified so that it no longer updates every frame, which fixes the buttons correctly staying highlighted after selection. The refresh method is only called when the list has changed e.g. in `BuyItemsScreen` and `SellItems` after you have bought or sold an item.